### PR TITLE
dev-v2.8 - gatekeeper 3.13.0

### LINKED ISSUE #41781: rancher-gatekeeper k8s 1.27 chart support
This issue is to implement the chart changes & perform the testing required to upgrade to k8s 1.27. Please perform below steps. 

- [x] Check whether the chart has any references to the below deprecated features of k8s 1.27. You can find information about deprecated APIs here on [Kubernetes Deprecation guide](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-27).

**storage.k8s.io/v1beta1** API version of **CSIStorageCapacity**
 

- [x] Update kube-version annotation in Chart.yaml. catalog.cattle.io/kube-version: '>= 1.23.0-0 < 1.28.0-0'

Note: If this is an upstream chart and if the upstream chart doesn't support k8s 1.27 then do not update the kube-version annotation or update the k8s version accordingly.

- [x] Perform below tests

- Fresh install/uninstall/upgrade/downgrade of the charts on K8s 1.27 provisioned on Rancher. Test the chart functionality.
(Please refer to the https://github.com/rancher/rancher/issues/41395 to know what testing was done for k8s 1.26 upgrade for your chart)
- Provision rancher on k8s 1.26 or lesser
- Install chart on this cluster
- Upgrade this cluster in rancher to k8s 1.27
- Make sure the existing chart and the app is functional after cluster upgrade

### PR DESCRIPTION
**Pull request by diogoasouza**
fixes https://github.com/rancher/rancher/issues/41781

depends on https://github.com/rancher/image-mirror/pull/446

### CODE REVIEW

**Review by rohitsakala — CHANGES_REQUESTED** [review]
What about assets and charts folder ?

File: packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch [hunk 1]
```diff
@@ -31,12 +32,12 @@
...
          - apply
          - -f
-@@ -120,11 +108,9 @@
+@@ -123,11 +111,9 @@
            {{- end }}
            {{- toYaml .Values.crds.securityContext | nindent 10 }}
        {{- with .Values.crds }}
-+      nodeSelector: {{ include "linux-node-selector" . | nindent 8 }}
-+      tolerations: {{ include "linux-node-tolerations" . | nindent 8 }}
++      nodeSelector: { { include "linux-node-selector" . | nindent 8 } }
```
Comment (rohitsuse) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch]: The curly brackets have space in between them. ? It shouldn't right ?

File: packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch [hunk 2]
```diff
@@ -31,12 +32,12 @@
...
++      tolerations: { { include "linux-node-tolerations" . | nindent 8 } }
```
Comment (rohitsuse) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch]: The curly brackets have space in between them. ? It shouldn't right ?

File: packages/rancher-gatekeeper/generated-changes/patch/values.yaml.patch [hunk 3]
```diff
@@ -1,44 +1,41 @@
...
--  release: v3.12.0
+-  release: v3.13.0
 +images:
 +  gatekeeper:
 +    repository: rancher/mirrored-openpolicyagent-gatekeeper
-+    tag: v3.12.0
-+  gatekeepercrd:
-+    repository: rancher/mirrored-openpolicyagent-gatekeeper-crds
-+    tag: v3.12.0
++    tag: v3.13.0
```
Comment (rohitsuse) [inline packages/rancher-gatekeeper/generated-changes/patch/values.yaml.patch:14]: Did we remove rancher/mirrored-openpolicyagent-gatekeeper-crds image from values.yaml ? Was this intential ?
```
gatekeepercrd:
+    repository: rancher/mirrored-openpolicyagent-gatekeeper-crds
+    tag: v3.12.0
```

**Review by rohitsakala — CHANGES_REQUESTED** [review]
posted some doubts as comments.

**Review by rohitsakala — APPROVED** [review]

**Pull request merged**